### ISSUE TITLE
[FEATURE] support owned object in ThirdPossession

### ISSUE THREAD

**Issue by blegaut** [RosaeNLG/rosaenlg#164]
**Is your feature request related to a problem? Please describe.**
I would like to be able to use an object with a referential expression as owned parameter in thirdPossession

**Describe the solution you'd like**
I would like the following code

```
- var BAGUE = {};
mixin bague_ref(obj, params)
  | #[+value('bague', {represents:BAGUE, det:'DEFINITE'})]
- BAGUE.ref = bague_ref;

- var LARGEUR = {};
mixin largeur_ref(obj, params)
  | #[+value('largeur', {represents:LARGEUR, det:'DEFINITE', adj:'impressionnant', adjPos:'BEFORE'})]
- LARGEUR.ref = largeur_ref;
- setRefGenderNumber(LARGEUR, 'F', 'S');


p #[+thirdPossession(BAGUE, LARGEUR)] / #[+thirdPossession(BAGUE, LARGEUR)]
```

 to produce
`<p>L'impressionnante largeur de la bague / son impressionnante largeur</p>`
when we got now:
`<p>L'impressionnante largeur de la bague / sa [object Object]</p>`

**Describe alternatives you've considered**

I did a slight modification in `thirdPossessionRefTriggered` in rosaenlg/src/LanguageFrench.ts. 
![image](https://user-images.githubusercontent.com/92172115/221709894-d24bce87-ba96-4e66-a6e9-c7451e9bc4d5.png)

Happy to make a PR 
**Additional context**
Add any other context about the feature request here.

**Comment by ludans** [comment]
Checking `this.valueManager.value(owned, Object.assign({}, params, { det: '' }));`:
- `owned` is an object: `LARGEUR`
- using `value` (as in `this.valueManager.value`) on `LARGEUR` will actually trigger its reference
- force removing the determiner here (using `det: ''`) will not work: the object has no determiner. The original `det:'DEFINITE'` will remain.

My test with this code generated _l'impressionnante largeur de la bague / sa l'impressionnante largeur_ - which is not good.

To be able to do it, a way could be to have more structured objects in mixins:
```
- 
  LARGEUR.ref = {
    noun: 'largeur',
    represents:LARGEUR, 
    det: 'DEFINITE', 
    adj:'impressionnant', 
    adjPos:'BEFORE',
  };
```
instead of
```
mixin largeur_ref(obj, params)
  | #[+value('largeur', {represents:LARGEUR, det: 'DEFINITE', adj:'impressionnant', adjPos:'BEFORE'})]
- LARGEUR.ref = largeur_ref;
```


PS:
```
spy.appendPugHtml(det);
this.helper.insertSeparatingSpaceIfRequired();
```
is better than

```
spy.appendPugHtml(`$ {det} `);
```
as it allows to track more easily where the spaces come from (from the user or added by RosaeNLG).

**Comment by blegaut** [comment]
solved wiht rosae 4.0.0. Here a example

```
- var BAGUE = {};
mixin bague_ref(obj, params)
  | #[+value('bague', {represents:BAGUE, det:'DEFINITE'})]
  - setRefGenderNumber(BAGUE, 'F', 'S');
- BAGUE.ref = bague_ref;

- var LARGEUR = {};
mixin largeur_ref(obj, params)
  | #[+value('largeur', {...{represents:LARGEUR, det:'DEFINITE', adj:'grande', adjPos:'BEFORE'}, ...params})]
  - setRefGenderNumber(LARGEUR, 'F', 'S');
- LARGEUR.ref = largeur_ref;
- setRefGenderNumber(LARGEUR, 'F', 'S');


p #[+thirdPossession(BAGUE, LARGEUR)] / #[+thirdPossession(BAGUE, LARGEUR)]
```
will produce
`<p>La grande largeur de la bague / sa grande largeur</p>`
